reactioncommerce/reaction
# Record plugin versions in DB and show in Shop panel

Resolves #4874 
Impact: **minor**  
Type: **feature**

## Issue
Have some standard way to record versions of plugins and make them visible to an admin so that you know what code is deployed

## Solution
Plugins can add a "version" key to `register.js` and they will be stored in the db and displayed in the shop panel

## Breaking changes
None


## Testing
1. Add a new plugin with a `register.js` that looks like this:
```
import Reaction from "/imports/plugins/core/core/server/Reaction";

Reaction.registerPackage({
  label: "Plugin-Versions",
  name: "reaction-plugin-versions",
  version: "1.0",
  icon: "fa fa-vines",
  autoEnable: true,
  registry: [{
    provides: ["dashboard"],
    label: "Plugin Versions",
    description: "Plugin Versions",
    icon: "fa fa-vines",
    template: "PluginVersionsSettings"
  }]
});
```
2. Restart reaction
3. Go to the dashboard. shop settings and click on "plugin version" and observe that the version number is available there.
